compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -585,8 +585,9 @@ object Trees {
 
   /** A tree representing inlined code.
    *
-   *  @param  call      Info about the original call that was inlined.
-   *                    Only a reference to the toplevel class from which the call was inlined.
+   *  @param  call      Info about the original call that was inlined
+   *                    Until PostTyper, this is the full call, afterwards only
+   *                    a reference to the toplevel class from which the call was inlined.
```

Comment:
Would it be possible to keep the original call somewhere? This information is important:
- for incremental compilation, because the parameters of an inlined call may introduce dependencies that are not visible in the resulting inlined code
- for the IDE, because the user works with trees before inlining.

Reply:
A potential problem is that this could increase bytecode size significantly. E.g. looking at the trace calls in the compiler, which surround large bodies of code. If we keep the call that code would be duplicated in the Tasty tree.

Currently, we get the calls for trees that are in an edit buffer, but not for others. So findReferences would only show those references that survive in the inlined code, but not other references that are not inlined. 

We should think a bit more how we can reconcile these concerns.

Reply:
Regarding bytecode size: if we have several nested trace-like calls the blowup could be exponential. You can see this sometimes when you print at phase frontend without -Yshow-no-inlines.